### ISSUE TITLE
Use multiple configurations of i18n in express server

### ISSUE THREAD

**Issue by mebibou** [mashpie/i18n-node#149]
I'd like to have different i18n for handling different scopes, i.e. I would have my default configurations files that would handle default locales, and then other files to handle errors:
`conf/locals/en.json`:

```
{
  "default message": "default message"
}
```

and `conf/errors/en.json`:

```
{
  "error message": "error message"
}
```

or maybe having a subfolder for the error messages... any idea welcomed! :)

**Comment by mashpie** [comment]
with `i18n.init` as middleware you should be able to plug different configs into different routes.

**Comment by mebibou** [comment]
@mashpie okay thanks but how do I do that? couldn't find it in the examples or tests

**Comment by chambrid** [comment]
Any updates/answers on this issue? I think its similar to the question I asked in #251

**Comment by mashpie** [comment]
using the `register` option should help. Inspiration in https://github.com/mashpie/i18n-node/blob/master/test/i18n.configureRegister.js and https://github.com/mashpie/i18n-node/blob/master/test/i18n.configureApi.js and https://github.com/mashpie/i18n-node/blob/master/test/i18n.init.js 

Just a quick thought of what might work (I can't test now):

```js
const express = require('express');
const i18nMessages = require("i18n");
const i18nErrors = require("i18n");

i18nMessages.configure({
    locales: ['en', 'de'],
    directory: __dirname + '/messages/locales',
    api: {
        '__': '__m',
        '__n': '__mn'
    }
});

i18nErrors.configure({
    locales: ['en', 'de'],
    directory: __dirname + '/errors/locales',
    api: {
        '__': '__e',
        '__n': '__en'
    }
});

app.use(i18nMessages.init);
app.use(i18nErrors.init);

app.get('/test', (req, res) => {
  let msg = req.__m('default message'); // should translate messages
  let err = req.__e('error message'); // should translate errors

  res.send(/* ...whatsoever... */);
});
```

**Comment by HughePaul** [comment]
> const i18nMessages = require("i18n");
> const i18nErrors = require("i18n");

Surely both i18nMessages and i18nErrors would be assigned the same object from the cached require?

**Comment by chambrid** [comment]
I found the following fork that resolves this problem for me: https://github.com/jeresig/i18n-node-2

**Comment by mashpie** [comment]
working example with cleaning require cache https://github.com/mashpie/i18n-node/blob/master/test/i18n.configureMustache.js

still there will be a factory / constructor option sooner

**Comment by mashpie** [comment]
Instance feature is released as of 0.12.0 with some extra sugar (shortcut to i18n.configure on creation):

```js
const { I18n } = require("i18n");

const i18n = new I18n({
    locales:['en', 'de'],
    directory: __dirname + '/locales'
});
```

see https://github.com/mashpie/i18n-node/tree/master#as-instance for details
